vmngclient/api/template_api.py

```diff
@@ -531,6 +540,27 @@ def generate_feature_template_payload(
 
         return payload
 
+    def validate_device_model(self, template: FeatureTemplate) -> bool:
+        """Verify if selected template can be used with provided device model"""
+        template_type = self._get_feature_template_types().filter(name=template.type).single_or_default()
+
+        available_devices_for_template = [device["name"] for device in template_type.device_models]
+        provided_device_models = [
+            dev_mod.value if type(dev_mod) is DeviceModel else dev_mod for dev_mod in template.device_models
+        ]
```

Comment:
placeholder if `device_model` attribute could be converted into only `DeviceModel` object (not `str`)